tests/test_dataset.py

```diff
@@ -229,6 +289,23 @@ def test_records_limit_and_sample(spark_context):
     assert records.count() == 5
 
 
+def decode(obj):
+    value = obj.getvalue()
+    return [json.loads(value)]
+
+
+@pytest.mark.slow
+def test_records_selection(spark_context):
+    bucket_name = 'test-bucket'
+    store = InMemoryStore(bucket_name)
+    key = 'dir1/subdir1/key1'
+    value = '{"a": {"b": { "c": "value"}}}'
+    store.store[key] = value
+    dataset = Dataset(bucket_name, ['dim1', 'dim2'], store=store).select(field='a.b.c')
+    records = dataset.records(spark_context, decode=decode)
+    assert records.collect() == [{'field': 'value'}]
```

Comment:
Can we add another few lines here to test chaining another select to `dataset` and calling `records()` again?